### PR TITLE
#456 Check for 2-4bit PNGs and skip the native writer if that's what comes up

### PR DESCRIPTION
**Pull request by dvntucker**
This is a bit of a corner case, but 2 and 4bit PNGs don't work with Native JAI. When using PNG8 with GeoServer sometimes the PNGs get encoded as 4bit PNGs as an optimization.

The solution is a bit hacky, but not too far off of how GeoServer/GeoTools handle this incompatibility. In GeoTools the native SPI is found by name and asked if it can support the RenderedImage being handled. That was a bit heavy weight for GWC, so in this case I just check for 2/4 bit PNGs and if the writer returned matches the native version (just going by class name, since we can't statically link against that class for an instanceof check) it will skip it and move on to the next writer in the list.

GS/GT's handling of the same edge case can be found in PNGMapResponse/ImageWorker respectively.

### CODE REVIEW

File: geowebcache/core/src/test/java/org/geowebcache/layer/wms/MetaTileTest.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void test1MetaTile() throws Exception {
                 30);
         
         long[] gridPos = { 0, 0, 0 };
-        
+
         //int[] gridBounds, int[] tileGridPosition, int metaX, int metaY
```
Comment (smithkm) [inline geowebcache/core/src/test/java/org/geowebcache/layer/wms/MetaTileTest.java]: Unnecessary whitespace change, but minor so we can live with it.

File: geowebcache/core/src/test/java/org/geowebcache/mime/ImageMimeTest.java [hunk 2]
```diff
@@ -114,4 +120,17 @@ public void testIsCompatileFailsOnXMLMime() {
...
             pngMimeType.isCompatible("application/vnd.ogc.se_xml"));
     }
+
+    @Test
+    public void test4BitPNG() throws IOException, URISyntaxException {
+        URL url = this.getClass().getResource("/images/4bit.png");
+        RenderedImage tile = ImageIO.read(new File(url.toURI()));
+        ImageWriter writer = ImageMime.png8.getImageWriter(tile);
+        //Not a great test, but make sure our writer is the pure java one, not the native version
+        //since the native version can't handle 4bit pngs.
```
Comment (smithkm) [inline geowebcache/core/src/test/java/org/geowebcache/mime/ImageMimeTest.java]: Agreed. This is probably the best that's practical, even if it's far from ideal.  All I can think of as an alternative would be wrapping the `javax.imageio.ImageIO.getImageWritersByFormatName(internalName);` call in a method that could be overridden for testing and then providing mocks, which still has the problem of providing a mock for a class that may or may not exist where the specific class name is the important detail.
